### PR TITLE
Add add_rule() method to widget.

### PR DESCRIPTION
**Pull request by willrogers**
@nickbattam @mfurseman

### CODE REVIEW

File: opimodel/widgets.py [hunk 1]
```diff
@@ -104,6 +105,9 @@ def set_font(self, font):
         """
         self.font = font
 
+    def add_rule(self, rule):
```
Comment (nickbattam) [inline opimodel/widgets.py:108]: Need to add a docstring.

**Comment by nickbattam** [comment]
This change means every widget will have `<rules/>` (or `<rules></rules>`) added to the generated XML.
It would be neater to only render if the list is non-empty.

File: test/test_rules.py [hunk 2]
```diff
@@ -1,16 +1,15 @@
 from opimodel import rules, colors
 
 
-def test_empty_RulesNode(widget, get_renderer):
+def test_empty_RulesNode_does_not_create_rules_tag(widget, get_renderer):
```
Comment (nickbattam) [inline test/test_rules.py]: This test needs updating now the rules list is not API, or a second test adding to check "no rules node rendered when no rules added"

Comment (willrogers) [reply test/test_rules.py]: Do you mean I should remove line 5?

Comment (nickbattam) [reply test/test_rules.py]: yes

**Pull request merged**